src/lib/secret.ts

```diff
@@ -1,22 +1,36 @@
 export class Secret {
   // A fauna key, like `fn1234`.
   key: string;
+  // Do we allow database scope?
+  allowDatabase: boolean;
   // A database scope, like `["foo", "bar"]`
   databaseScope: string[];
 
-  constructor(opts: { key: string; databaseScope?: string[] }) {
+  constructor(opts: {
+    key: string;
+    allowDatabase: boolean;
+    databaseScope?: string[];
+  }) {
     this.key = opts.key;
+    this.allowDatabase = opts.allowDatabase;
     this.databaseScope = opts.databaseScope ?? [];
   }
 
+  static parseFlag(key: string) {
+    if (key.length === 0) {
+      throw new Error("Secret cannot be empty");
+    }
+    return new Secret({ key, allowDatabase: !key.includes(":") });
+  }
+
   static parse(key: string) {
     if (key.length === 0) {
       throw new Error("Secret cannot be empty");
     }
     if (key.includes(":")) {
       throw new Error("Secret cannot be scoped");
     }
```

Comment:
my understanding here is that we wouldn't allow scoped secrets in the `.fauna-shell` file.  I think that is fine, is backwards incompatible I believe though so could be a good thing for us to be aware of.